src/sdk/refundPegin.ts

```diff
@@ -0,0 +1,48 @@
+import type { Quote } from '../api'
+import type { FlyoverSDKContext } from '../utils/interfaces'
+import { isPeginRefundable } from './isPeginRefundable'
+import { getRawTxWithoutWitnesses } from '../bitcoin/utils'
+import pmtBuilder from '@rsksmart/pmt-builder'
+
+export interface RefundPeginParams {
+  quote: Quote
+  providerSignature: string
+  userBtcTransactionHash: string
+  flyoverContext: FlyoverSDKContext
+}
+
+export async function refundPegin (
+  params: RefundPeginParams
+): Promise<string> {
+  const { quote, providerSignature, userBtcTransactionHash, flyoverContext } = params
+
+  const isRefundable = await isPeginRefundable({
+    quote,
+    providerSignature,
+    btcTransactionHash: userBtcTransactionHash,
+    flyoverContext
+  })
+
+  if (!isRefundable.isRefundable) {
+    throw isRefundable.error
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const btcDataSource = flyoverContext.btcConnection!
+
+  const btcRawTxWithoutWitnesses = await getRawTxWithoutWitnesses(userBtcTransactionHash, btcDataSource)
+
+  const block = await btcDataSource.getBlockFromTransaction(userBtcTransactionHash)
+  const partialMarkleTree = pmtBuilder.buildPMT(block.transactionHashes, userBtcTransactionHash)
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const txResult = await flyoverContext.lbc!.registerPegin({
```

Comment:
just a comment (not sure if we talked about this or not), but if you prefer you can use the assertTruthy utility instead of the assertion (up to you)

Reply:
Thanks, in this case I prefer this one because, based on the flow of the app, a this point I'm sure the lbc is defined because it was validated in Flyover before calling this function.